raft_test.go

```diff
@@ -2059,18 +2059,18 @@ func TestRaft_LeadershipTransferLeaderRejectsClientRequests(t *testing.T) {
 		l.requestConfigChange(configurationChangeRequest{}, 100*time.Millisecond),
 	}
 	futures = append(futures, l.LeadershipTransfer())
-	select {
-	case <-l.leadershipTransferCh:
-	default:
-	}
```

Comment:
This was the core of the problem. This "read from channel" would race with the one in `leaderLoop`, and if this one won the race the test would block until the timeout.